languages/python/oso/oso/oso.py

```diff
@@ -38,3 +38,24 @@ def is_allowed(self, actor, action, resource) -> bool:
             return True
         except StopIteration:
             return False
+
+    def get_allowed_actions(self, actor, resource, allow_wildcard=False):
+        # Get allowed actions on the resource
+        results = self.query_rule("allow", actor, Variable("action"), resource)
+        actions = set()
+        for result in results:
+            action = result.get("bindings").get("action")
+            if type(action) == Variable:
+                # TODO: is this the correct behavior?
```

Comment:
Yes. Although I would change the error message slightly. Maybe:

The result of `get_allowed_actions()` contained an "unconstrained" action that could represent any action, but `allow_wildcard` was set to true. To fix, set `allow_wildcard` to true and compare with the `"*"` string.

I _think_ if someone gets back an unbound this is probably what they are looking to achieve? I like it being off-by-default so that people need to "learn" this behaviour instead of being surprised by the result, however.

Reply:
Yeah that's a good suggestion, done.